php_phongo.c

```diff
@@ -144,38 +144,15 @@ zend_class_entry* phongo_exception_from_mongoc_domain(uint32_t /* mongoc_error_d
 			return php_phongo_connectiontimeoutexception_ce;
 		case MONGOC_ERROR_CLIENT_AUTHENTICATE:
 			return php_phongo_authenticationexception_ce;
-
 		case MONGOC_ERROR_COMMAND_INVALID_ARG:
 			return php_phongo_invalidargumentexception_ce;
-
-		case MONGOC_ERROR_STREAM_INVALID_TYPE:
-		case MONGOC_ERROR_STREAM_INVALID_STATE:
 		case MONGOC_ERROR_STREAM_NAME_RESOLUTION:
 		case MONGOC_ERROR_STREAM_CONNECT:
-		case MONGOC_ERROR_STREAM_NOT_ESTABLISHED:
 			return php_phongo_connectionexception_ce;
-		case MONGOC_ERROR_CLIENT_NOT_READY:
-		case MONGOC_ERROR_CLIENT_TOO_BIG:
-		case MONGOC_ERROR_CLIENT_TOO_SMALL:
-		case MONGOC_ERROR_CLIENT_GETNONCE:
-		case MONGOC_ERROR_CLIENT_NO_ACCEPTABLE_PEER:
-		case MONGOC_ERROR_CLIENT_IN_EXHAUST:
-		case MONGOC_ERROR_PROTOCOL_INVALID_REPLY:
-		case MONGOC_ERROR_PROTOCOL_BAD_WIRE_VERSION:
-		case MONGOC_ERROR_CURSOR_INVALID_CURSOR:
-		/*case MONGOC_ERROR_PROTOCOL_ERROR:*/
-		case MONGOC_ERROR_BSON_INVALID:
-		case MONGOC_ERROR_MATCHER_INVALID:
-		case MONGOC_ERROR_NAMESPACE_INVALID:
-		case MONGOC_ERROR_GRIDFS_INVALID_FILENAME:
-			return php_phongo_runtimeexception_ce;
-		case MONGOC_ERROR_QUERY_FAILURE:
-		case MONGOC_ERROR_QUERY_COMMAND_NOT_FOUND:
-		case MONGOC_ERROR_QUERY_NOT_TAILABLE:
-		case MONGOC_ERROR_COLLECTION_INSERT_FAILED:
-			return php_phongo_serverexception_ce;
 	}
 	switch (domain) {
+		case MONGOC_ERROR_COMMAND:
+			// TODO: return php_phongo_commandexception_ce after PHPC-1089 is merged
```

Comment:
Since #771 is also open, does one of these PRs depend on the other? Based on this comment, it looks like you might want to get this PR merged and then rebase #771 afterwards.

Reply:
Yep, that sounds like a good plan.